mat64/dense.go

```diff
@@ -65,6 +65,11 @@ type Dense struct {
 	capRows, capCols int
 }
 
+// NewDense creates a new matrix of type *Dense with dimensions r and c. If the mat argument is nil,
+// a new data slice is allocated.
+// Data slice is of Row-major order, i.e., the (i, j) element in matrix should be at (i*c + j)-th
+// position in mat.
+// Also note that NewDense(0, 0, nil) can be used for undetermined size matrix initialization.
```

Comment:
1. My thought was to add an example of using Matrix to do something interesting. Like, ExampleTrace, where two matrices are multiplied and the trace is taken (or something). It would be added documentation.
2. From the package-level documentation "Note that in all interfaces that assign the result to the receiver, the receiver must be either the correct dimensions for the result or a zero-sized matrix. In the latter case, matrix data is allocated and stored in the receiver."
NewDense(0, 0, mat) is a zero-sized matrix, so its dimensions will be adjusted accordingly.